L1Trigger/L1TTrackMatch/plugins/L1TrackerEtMissProducer.cc

```diff
@@ -162,23 +194,20 @@ void L1TrackerEtMissProducer::produce(edm::Event& iEvent, const edm::EventSetup&
     if (nPS < nPSStubsMin_)
       continue;
 
+    numqualitytracks++;
+
     if (!displaced_) {  // if displaced, deltaZ = 3.0 cm, very loose
       // construct deltaZ cut to be based on track eta
-      if (fabs(eta) >= 0 && fabs(eta) < 0.7)
-        deltaZ_ = 0.4;
-      else if (fabs(eta) >= 0.7 && fabs(eta) < 1.0)
-        deltaZ_ = 0.6;
-      else if (fabs(eta) >= 1.0 && fabs(eta) < 1.2)
-        deltaZ_ = 0.76;
-      else if (fabs(eta) >= 1.2 && fabs(eta) < 1.6)
-        deltaZ_ = 1.0;
-      else if (fabs(eta) >= 1.6 && fabs(eta) < 2.0)
-        deltaZ_ = 1.7;
-      else if (fabs(eta) >= 2.0 && fabs(eta) <= 2.4)
-        deltaZ_ = 2.2;
+      for (unsigned int reg = 0; reg < etaRegions_.size(); reg++) {
+        if (fabs(eta) >= etaRegions_[reg] && fabs(eta) < etaRegions_[reg + 1]) {
```

Comment:
This imply accessing `etaRegions_[etaRegions_.size()]`: the last bin should be treated differently